ui/src/containers/PrivateRoute.js

```diff
@@ -3,16 +3,22 @@ import { useSelector } from 'react-redux';
 import { Route } from 'react-router-dom';
 
 const PrivateRoute = ({ component, ...rest }) => {
-  const authenticated = useSelector(state => !!state.oidc.user);
-  const userManager = useSelector(state => state.config.userManager);
+  const authenticated = useSelector((state) => !!state.oidc.user);
+  const userManager = useSelector((state) => state.config.userManager);
+  const isSaltAPIAuthenticated = useSelector((state) => state.login.salt);
 
   if (!authenticated) {
+    // Go to Dex Login Form if not authenticated
     userManager.signinRedirect({
       data: { path: window.location.pathname },
-    }); //Go to Dex Login Form if not authenticated
+    });
     return null;
-  } else {
+  } else if (isSaltAPIAuthenticated) {
+    // To make sure that Salt API is authenticated, before load the components
+    // Otherwise we may get 401 Unauthorized
```

Comment:
That's a band-aid which is "covering" the issue but not fixing it. Actually, sagas which require Salt API shouldn't be called before we are authenticated. But components could very well be shown before that. And since most of our data doesn't come from Salt anyway, it makes a lot of sense to show it before we are authenticated to SaltAPI.

Reply:
You're absolutely right! There is nothing stopping us load the component. 
We should use `yield race` to wait for Salt Authenticated success action. 
It's great to discover this function. 

Thank you!